pymc3/distributions/distribution.py

```diff
@@ -386,23 +387,37 @@ def random(self, point=None, size=None, **kwargs):
             not_broadcast_kwargs.update(**kwargs)
             if self.wrap_random_with_dist_shape:
                 size = to_tuple(size)
-                with _DrawValuesContextBlocker():
-                    test_draw = generate_samples(
-                        self.rand,
-                        size=None,
-                        not_broadcast_kwargs=not_broadcast_kwargs,
+                # We don't know anything about what self.rand returns so we
+                # have to try and make a test draw to check returned shape
+                try:
+                    with _DrawValuesContextBlocker():
+                        test_draw = self.rand(point=point, size=size)
+                        single_draw_shape = test_draw.shape
+                except Exception:
+                    # Something went wrong so we cannot know the true
+                    # single_draw_shape, and set it to an empty tuple. This
+                    # means that we must trust that self.shape is correct
+                    single_draw_shape = tuple()
+                if single_draw_shape[:len(size)] == size and point is not None:
+                    # The test shape has the size prepend and the supplied
+                    # point is not empty. This can happen when sampling from
+                    # the posterior predictive so we must take special care
```

Comment:
Will you please say why we must take special care and/or what is the thing that we need to take special care about?
Thanks!